Clarify Checkout view overrides require a rebuild on Umbraco 17+ (#8277)
umbraco/UmbracoDocs: commerce-add-ons/packages/checkout/customize-checkout.md

Umbraco Commerce Checkout is a free and open-source add-on package for Umbraco Commerce. It is possible to amend the default behavior to customize the checkout to your needs.

## Overriding the checkout Views

> Checkout's Views are precompiled into the package. To customize one, add your own copy at the same path under `Views/UmbracoCommerceCheckout`, and make sure your site compiles Views at build time so your copy is used instead of the packaged one.

Reviewer: ```suggestion
To customize one, add your copy to the same path under Views/UmbracoCommerceCheckout. Ensure your site compiles Views at build time so your copy is used instead of the packaged one.
```

2. Remove the `RazorCompileOnBuild` and `RazorCompileOnPublish` lines above (or set them to `true`).

This matches the configuration Umbraco requires for [Production runtime mode](https://docs.umbraco.com/umbraco-cms/fundamentals/setup/server-setup/runtime-modes).
{% endhint %}

Once your site compiles Views at build time, override a checkout step as follows:

1. Copy the equivalent [files and partials](https://github.com/umbraco/Umbraco.Commerce.Checkout/tree/main/src/Umbraco.Commerce.Checkout/Views/UmbracoCommerceCheckout).
2. Add them to `Views/UmbracoCommerceCheckout` in your project directory. It might be necessary to create the folder first.
3. Make your changes.
4. Rebuild and restart your site, then verify that the change is displayed correctly.

## Alternative: use a Template

Instead of overriding the Views, you can assign a **Template** to the checkout page in the backoffice and place your markup there. Checkout will use the Template instead of its built-in View.

> With this approach the Template receives the checkout page's `IPublishedContent` as its model - not the view model the packaged Views use - so you build that step's markup yourself against the Umbraco Commerce APIs.

Reviewer: ```suggestion
With this approach, the Template receives the checkout page's IPublishedContent as its model—not the packaged Views' view model.

You then build that step's markup yourself against the Umbraco Commerce APIs.
```
